core-bundle/tests/Image/IntegrationTest.php

```diff
@@ -0,0 +1,1688 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Tests\Image;
+
+use Contao\Config;
+use Contao\Controller;
+use Contao\CoreBundle\Framework\Adapter;
+use Contao\CoreBundle\Image\ImageFactory;
+use Contao\CoreBundle\Image\LegacyResizer;
+use Contao\CoreBundle\Image\PictureFactory;
+use Contao\CoreBundle\Image\Studio\Studio;
+use Contao\CoreBundle\Tests\TestCase;
+use Contao\FilesModel;
+use Contao\FrontendTemplate;
+use Contao\Image\PictureGenerator;
+use Contao\Image\ResizeCalculator;
+use Contao\ImagineSvg\Imagine as ImagineSvg;
+use Contao\LayoutModel;
+use Contao\Model\Registry;
+use Contao\PageModel;
+use Contao\System;
+use Contao\Template;
+use Imagine\Gd\Imagine as ImagineGd;
+use Psr\Log\NullLogger;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBag;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Webmozart\PathUtil\Path;
+
+class IntegrationTest extends TestCase
```

Comment:
I don't like the class name. "Integration test" is a type of test and says nothing about what the class actually does.

Reply:
See a6e0e61a452ad416468102baf78d9917176528f9

I called it `FigureBuilderIntegrationTest` for now and put it in the `Studio` namespace. Because basically the new `Controller::addImageToTemplate` implementation uses the `FigureBuilder` (+ some additional BC stuff) to do its work.